chainstate/src/detail/ban_score.rs

```diff
@@ -61,18 +60,19 @@ impl BanScore for BlockError {
             BlockError::TxIndexConstructionError(_) => 100,
             BlockError::PoSAccountingError(err) => err.ban_score(),
             BlockError::EpochSealError(err) => err.ban_score(),
-            BlockError::BlockHeightTooBig(_) => 0,
+            BlockError::BlockTooDeepToInvalidate(_) => 0,
+            BlockError::BestChainCandidatesError(_) => 0,
+            BlockError::BlockNotFound(_) => 0,
 
             BlockError::BestBlockIdQueryError(_) => 0,
+            BlockError::BestBlockIndexQueryError(_) => 0,
             BlockError::IsBlockInMainChainQueryError(_, _) => 0,
-            BlockError::BlockIdTreeTopQueryError(_, _) => 0,
             BlockError::BlockIndexQueryError(_, _) => 0,
+            BlockError::BlockIndicesForBranchQueryError(_) => 0,
+            BlockError::MinHeightForReorgQueryError(_) => 0,
```

Comment:
If all these errors have to do with invalidation, perhaps we need another error type and a submodule that handles invalidation.